public/r2/v1/swagger/swagger.json

```diff
@@ -771,6 +775,52 @@
                     }
                 }
             }
+        },
+        "/rulesets/validate": {
```

Comment:
If we're validating the full RuleSet, why not have the validation process apart of the normal POST and PUT operations for the mapping and rollup roles.

For example, when you do a PUT on a mapping rules 
`PUT /namespaces/{namespaceID}/mapping-rules`
 
It can return a `HTTP 400: Mapping rule is invalid because {error}`

Reply:
Hey @saminzadeh I've actually updated the diff to include validation for namespace as well so things have changed around. I believe RuleSets are validated on all POST and PUT operations dealing with rules already. The motivation for this endpoint is to:

1) Provide a validation endpoint for proposed changes to a namespace (ie given a namespace name and its updated RuleSets - validate them). This will be used in CI to validate changes to rules when users have diffs created for their rule changes

2) Have a way to validate new changes while the user is making them from the UI (without them having to hit create/update.